stubs/mypy-extensions/mypy_extensions.pyi

```diff
@@ -21,12 +21,30 @@ class _TypedDict(Mapping[str, object], metaclass=abc.ABCMeta):
     def __delitem__(self, k: NoReturn) -> None: ...
 
 def TypedDict(typename: str, fields: dict[str, type[Any]], total: bool = ...) -> type[dict[str, Any]]: ...
-def Arg(type: _T = ..., name: str | None = ...) -> _T: ...
-def DefaultArg(type: _T = ..., name: str | None = ...) -> _T: ...
-def NamedArg(type: _T = ..., name: str | None = ...) -> _T: ...
-def DefaultNamedArg(type: _T = ..., name: str | None = ...) -> _T: ...
-def VarArg(type: _T = ...) -> _T: ...
-def KwArg(type: _T = ...) -> _T: ...
+@overload
+def Arg(type: _T, name: str | None = ...) -> _T: ...
+@overload
+def Arg(name: str | None = ...) -> Any: ...
```

Comment:
> Also we should make sure that mypy still works with this change, it may special-case the definitions here.

I was hoping primer would answer that question for me :)

Reply:
Not sure anyone in primer uses these things though.

Reply:
Tests all passed on my machine using `MYPYPATH`, I'm now trying @hauntsaninja's way as well.

Reply:
Tests also all pass on my machine with the manual edit of `mypy/typeshed/stubs/mypy-extensions/mypy_extensions.pyi`.

I'm also happy for this to be reverted if mypy's CI starts breaking unexpectedly!

Reply:
No problem!